plugins/mcreator-core/blockly/js/extensions_procedure_deps.js

```diff
@@ -0,0 +1,225 @@
+            // Trigger the change only if a block is changed, moved, deleted or created
+            if ((changeEvent.type !== Blockly.Events.BLOCK_CHANGE ||
+                changeEvent.element !== 'field') &&
+                changeEvent.type !== Blockly.Events.BLOCK_CREATE &&
+                changeEvent.type !== Blockly.Events.BLOCK_MOVE) {
+                return;
+            }
+            const group = Blockly.Events.getGroup();
+            // Makes it so the block change and the unplug event get undone together.
+            Blockly.Events.setGroup(changeEvent.group);
+            const procedure = this.getFieldValue('procedure');
+            for (let i = 0; this.getField('name' + i); i++) {
+                const prevType = this.getInput('arg' + i).connection.getCheck();
+                let depType = null;
+                const depList = javabridge.getDependencies(procedure);
+                for (const dep of depList) {
+                    if (dep.getName() === this.getFieldValue('name' + i)) {
+                        depType = dep.getBlocklyType();
+                        if (depType === '')
+                            depType = [];
+                        break;
+                    }
+                }
+                // Set input checks from dependency type
+                this.getInput('arg' + i).setCheck(depType);
+                const newType = this.getInput('arg' + i).connection.getCheck();
+                // Fire change event if block existed earlier and previous input type was different
+                if (changeEvent.type === Blockly.Events.BLOCK_CHANGE &&
+                    JSON.stringify(prevType) !== JSON.stringify(newType)) {
+                    const inputCheckChange = new Blockly.Events.BlockChange(this, null, 'arg' + i, prevType, newType);
+                    inputCheckChange.run = function (forward) {
+                        const block = this.blockId && this.getEventWorkspace_().getBlockById(this.blockId);
+                        if (block)
+                            block.getInput(this.name).setCheck(forward ? this.newValue : this.oldValue);
+                    };
+                    Blockly.Events.fire(inputCheckChange);
+                }
+            }
+            Blockly.Events.setGroup(group);
+        });
+    });
+
+// Helper function to use in Blockly extensions that validate repeating fields' values meant to be unique
+// The nullValue function is used when mutator needs to set a valid value in the field right after its creation
+function uniqueValueValidator(fieldName) {
+    return function (newValue) {
+        for (let i = 0; this.sourceBlock_.getField(fieldName + i); i++) {
+            if (this.sourceBlock_.getFieldValue(fieldName + i) === newValue && (fieldName + i) !== this.name)
+                return null;
+        }
+        return newValue;
+    };
+}
+
+// Helper function to find first index not taken by repeating fields with a certain name

```

Comment:
It seems some value checking is going on, but this comment does not mention how value of the field is handled in this function